sources/kube_factory.go

```diff
@@ -94,10 +95,28 @@ func CreateKubeSources(uri string, options map[string][]string) ([]api.Source, e
 			return nil, err
 		}
 	}
+	kubeletHttps := defaultKubeletHttps
+	if len(options["kubeletHttps"]) >= 1 {
+		kubeletHttps, err = strconv.ParseBool(options["kubeletHttps"][0])
+		if err != nil {
+			return nil, err
+		}
+	}
 	glog.Infof("Using Kubernetes client with master %q and version %s\n", kubeConfig.Host, kubeConfig.Version)
 	glog.Infof("Using kubelet port %d", kubeletPort)
-	kubeletApi := datasource.NewKubelet()
-	kubePodsSource := NewKubePodMetrics(kubeletPort, nodesApi, newPodsApi(kubeClient), kubeletApi)
+
+	kubeletConfig := &kube_client.KubeletConfig{
+		Port:            uint(kubeletPort),
+		EnableHttps:     kubeletHttps,
+		TLSClientConfig: kubeConfig.TLSClientConfig,
```

Comment:
@luksa: how does the clientauth json file exactly look. i need to define ca, cert and key for accessing my protected kubelets on port 10250 over https (including --anonymous-auth=false). thanks a lot for your answer. the doc say nothing about the format of the json..